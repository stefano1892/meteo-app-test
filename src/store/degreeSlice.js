import { createSlice } from "@reduxjs/toolkit";

const initialState = {
    degree: "Celsius"
}

export const degreeSlice = createSlice({
    name: 'degreeSelection',
    initialState,
    reducers: {
        changeDegree: (state, action) => {
            state.degree = action.payload
        }
    }
})

export const { changeDegree }  = degreeSlice.actions;

export default degreeSlice.reducer;