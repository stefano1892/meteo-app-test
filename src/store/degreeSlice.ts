import { createSlice } from "@reduxjs/toolkit";

interface InitialStateTypes {
    degree: string
}


const initialState: InitialStateTypes = {
    degree: "Celsius"
}

export const degreeSlice = createSlice({
    name: 'degreeSelection',
    initialState,
    reducers: {
        changeDegree: (state, action) => {
            state.degree = action.payload
        }
    }
})

export const { changeDegree }  = degreeSlice.actions;

export default degreeSlice.reducer;