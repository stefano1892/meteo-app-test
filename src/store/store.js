import { configureStore } from '@reduxjs/toolkit'
import weaterReducer from './weatherSlice'
import degreeReducer from './degreeSlice'

export const store = configureStore({
  reducer: {
    pastResearch: weaterReducer,
    degreeSelection: degreeReducer
  },
})