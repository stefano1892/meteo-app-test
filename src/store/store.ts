import { configureStore } from '@reduxjs/toolkit'
import weaterReducer from './weatherSlice'
import degreeReducer from './degreeSlice'

export const store = configureStore({
  reducer: {
    pastResearch: weaterReducer,
    degreeSelection: degreeReducer
  },
})

export type AppSelector = ReturnType<typeof store.getState>

// () => number => ReturnType<() => number> = number