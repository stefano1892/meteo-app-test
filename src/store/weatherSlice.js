import { createSlice } from "@reduxjs/toolkit";

const initialState = {
    searches: []
}

export const weatherSlice = createSlice({
    name: 'pastResearch',
    initialState,
    reducers: {
        addResearch: (state, action) => {
            state.searches.push(action.payload)
            if(state.searches.length > 5) {
                state.searches.shift()
            }
        }
    }
})

export const { addResearch }  = weatherSlice.actions;

export default weatherSlice.reducer;