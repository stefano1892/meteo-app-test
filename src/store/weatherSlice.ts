import { createSlice } from "@reduxjs/toolkit";
import { ICity } from "../interfaces/ICity";

interface InitialStateTypes {
    searches: ICity[]
}

const initialState: InitialStateTypes = {
    searches: []
}

export const weatherSlice = createSlice({
    name: 'pastResearch',
    initialState,
    reducers: {
        addResearch: (state, action) => {
            state.searches.push(action.payload)
            if(state.searches.length > 5) {
                state.searches.shift()
            }
        }
    }
})

export const { addResearch }  = weatherSlice.actions;

export default weatherSlice.reducer;