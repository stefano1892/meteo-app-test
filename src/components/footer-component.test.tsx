import React from "react"
import FooterComponent from "./footer-component"
import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { store } from "../store/store"

describe('FooterComponent', () => {
    test('Snap Footer', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <FooterComponent />
            </Provider>
        )

        expect(asFragment()).toMatchSnapshot();
    })
})