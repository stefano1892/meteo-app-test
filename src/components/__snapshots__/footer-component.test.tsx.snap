// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent Snap Footer 1`] = `
<DocumentFragment>
  <div
    class="w-100 mt-3 footer-containter"
  >
    <div
      class="w-100 m-0 row"
    >
      <div
        class="col"
      >
        <div
          class="footer-links-container text-center"
        >
          <b
            class="footer-link-title"
          >
            Link utili
          </b>
          <div
            class="mt-3"
          >
            <a
              class="footer-link"
              href="#"
            >
              Home
            </a>
          </div>
          <div>
            <a
              class="footer-link"
              href="#"
            >
              Previsioni
            </a>
          </div>
          <div>
            <a
              class="footer-link"
              href="#"
            >
              Situazione
            </a>
          </div>
          <div />
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="footer-links-container text-center"
        >
          <b
            class="footer-link-title"
          >
            Novità
          </b>
          <div
            class="mt-3"
          >
            <a
              class="footer-link"
              href="#"
            >
              Webcam
            </a>
          </div>
          <div>
            <a
              class="footer-link"
              href="#"
            >
              News
            </a>
          </div>
          <div>
            <a
              class="footer-link"
              href="#"
            >
              Viabilità
            </a>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="footer-links-container text-center"
        >
          <b
            class="footer-link-title"
          >
            Contatti
          </b>
          <div
            class="mt-3"
          >
            <a
              class="footer-link"
              href="#"
            >
              Contatti
            </a>
          </div>
          <div>
            <a
              class="footer-link"
              href="#"
            >
              Business
            </a>
          </div>
          <div>
            <a
              class="footer-link"
              href="#"
            >
              Lavora con noi
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="registered-mark"
    >
      © 2024 Copyright: Brandname.com
    </div>
  </div>
</DocumentFragment>
`;
